AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -798,33 +796,6 @@ - (void)transitionLayoutWithSizeRange:(ASSizeRange)constrainedSize
   }
 }
 
-- (void)_completeLayoutCalculation
-{
-  ASDN::MutexLocker l(_propertyLock);
-  [self calculatedLayoutDidChange];
-
-  // We generate placeholders at measureWithSizeRange: time so that a node is guaranteed to have a placeholder ready to go.
-  // This is also because measurement is usually asynchronous, but placeholders need to be set up synchronously.
-  // First measurement is guaranteed to be before the node is onscreen, so we can create the image async. but still have it appear sync.
-  if (_placeholderEnabled && [self _displaysAsynchronously] && self.contents == nil) {
-    
-    // Zero-sized nodes do not require a placeholder.
-    CGSize layoutSize = (_layout ? _layout.size : CGSizeZero);
-    if (CGSizeEqualToSize(layoutSize, CGSizeZero)) {
-      return;
-    }
-
-    if (!_placeholderImage) {
-      _placeholderImage = [self placeholderImage];
```

Comment:
Layout Calculation = Layout Transition now. Basically both go kind of the same way and finally call `_pendingLayoutTransitionDidComplete` that executes this code block. To show the different routes of layout transitions (1. Layout calculation / measurement and 2. Layout animation) I created this graph. Hope this makes it all a bit more clearer:

![transition api flow](https://cloud.githubusercontent.com/assets/25288/16917679/7a628b92-4cb7-11e6-8605-37f76022c414.png)